subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c

```diff
@@ -95,7 +95,7 @@ static bool check_endpoints_freed(struct ipc_rpmsg_instance *rpmsg_inst)
 	for (size_t i = 0; i < NUM_ENDPOINTS; i++) {
 		rpmsg_ept = &rpmsg_inst->endpoint[i];
 
-		if (strcmp("", rpmsg_ept->name) != 0) {
+		if (rpmsg_ept->bound == true) {
```

Comment:
Tagging @jacksonc-xlnx

IUC this is slightly changing the semantics though.

- `rpmsg_ept->bound == true` means that the EPT is bounded on the remote and local endpoint
- `strcmp("", rpmsg_ept->name) != 0` means that the EPT was advertised by the local endpoint (but not necessarily bounded yet)

Reply:
This change was made due to firmware on network core I am working with. In my case net FW creates two endpoints and the main app connects to one or another endpoint (in each situation it uses only one endpoint). In such case I was not able to successfully call `ipc_service_close_instance(...)` when the `check_endpoints_freed(...)` was checking by names.

Reply:
Did you call the `.deregister_endpoint` function on both the endpoints before closing the instance down?

Reply:
Only on the one that is bonded. For the second one I would need to register it first to have any handle for deregister. I have not found another way to execute 'ipc_service_close_instance' (one it to register and deregister unused endpoint on application side and the second one is to check for freed not by name but by bounded flag). 
If that change is problematic we can solve it on our side by registering and deregistering ept.

Reply:
The other option is to add a `FORCE` parameter to `ipc_service_close_instance` to basically forcefully closing all the endpoints associated to that instance.

Reply:
I am not very familiar with IPC but in my opinion endpoints that are not bound should not block closing API because they are not used by API on that side. From my point of view checking freed by bound flag was most elegant. Especially if the endpoint was registered by the second side.

Reply:
@jacksonc-xlnx ping

Reply:
This all sounds reasonable - and that looks like a good change. I would say that it belongs in a separate commit to the other changes however, as it is seemingly unrelated.